addons/mail/static/src/core/web/thread_model_patch.js

```diff
@@ -24,4 +25,87 @@ patch(Thread.prototype, {
     get recipientsFullyLoaded() {
         return this.recipientsCount === this.recipients.length;
     },
+    closeChatWindow() {
+        const chatWindow = this.store.discuss.chatWindows.find((c) => c.thread?.eq(this));
```

Comment:
note unrelated to PR: should have chat window(s) relation on thread to simplify this